cpp/kiss_icp/cmake/CompilerOptions.cmake

```diff
@@ -22,8 +22,14 @@
 # SOFTWARE.
 function(set_global_target_properties target)
   target_compile_features(${target} PUBLIC cxx_std_17)
-  target_compile_options(${target} PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-fcolor-diagnostics>)
-  target_compile_options(${target} PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,GNU>:-fdiagnostics-color=always>)
+
+  set(GCC_WARNINGS_FLAGS -Wall -Wextra -Wpedantic -Werror)
```

Comment:
Since you suggested ` /WX` for the MSVC build, I think it's only appropriate to use `-Werror`  with the rest. Most folks find this too extreme, it can also be a bit annoying during development. But I'm generally all for it. Dropping both `-Werror`  and `/WX`  would also be an option, however.

Reply:
Yes you are right, is too annoying. But I feel like the code won't change much. We only must ensure that these are private options and don't get propohated to dependent targets